### PR TITLE
[BEAM-1695] Improve Python-SDK's programming guide

### PR DESCRIPTION
**Pull request by tibkiss**

### CODE REVIEW

**Comment by tibkiss** [comment]
R: @aaltay

**Comment by tibkiss** [comment]
Please make sure that this gets reviewed & merged together with the respective beam-site change:
https://github.com/apache/beam/pull/2738

I did my best to test it locally (by running a local site, replacing the github references to local), but I recommend checking the website after the upgrade to ensure correctness.

**Comment by aaltay** [comment]
retest this please

**Comment by aaltay** [comment]
Waiting for new staged version to become available after changes to snippets.

File: src/documentation/programming-guide.md [hunk 1]
```diff
@@ -95,6 +95,8 @@ public static void main(String[] args) {
 ```py
 # Will parse the arguments passed into the application and construct a PipelineOptions object.
 # Note that --help will print registered options.
+import sys
```
Comment (aaltay) [inline src/documentation/programming-guide.md]: There is no need to add this, to the snippet. `PipelineOptions()` does this automatically.

Comment (tibkiss) [inline src/documentation/programming-guide.md]: Good catch! Removed it.

**Comment by aaltay** [comment]
LGTM. Thank you @tibkiss

**Comment by aaltay** [comment]
@tibkiss could you do a rebase to resolve conflicts?

**Comment by aaltay** [comment]
(It seems like the whole repo is missing for some reason.)

**Pull request closed without merging**